models/unit/unit.go

```diff
@@ -353,19 +353,23 @@ func AllUnitKeyNames() []string {
 	return res
 }
 
-// MinUnitAccessMode returns the minial permission of the permission map
+// MinUnitAccessMode returns the minimum permission of the permission map
 func MinUnitAccessMode(unitsMap map[Type]perm.AccessMode) perm.AccessMode {
-	res := perm.AccessModeNone
+	res := perm.AccessModeWrite
```

Comment:
Shouldn't it be
```suggestion
	res := perm.AccessModeOwner
```
and letting the for loop scale the variable down?
Or do we pass an empty map as parameter at any point?

Reply:
Can we have units with `perm.AccessModeOwner`? 

I see here we set MaxPerm=`perm.AccessModeAdmin`: https://github.com/go-gitea/gitea/blob/23ae939ef3ef03848038372de21490725855b5f9/models/unit/unit.go#L212

But I don't know of any units that can have admin/owner access in practice.

Reply:
TBH, I have difficulty to understand the logic if I didn't read the chat history & PR description carefully.

And the new logic seems counterintuitive , could there be some detailed comments or some tests for this problem?

Reply:
Yeah, comment and tests will be helpful regardless. I’ll work on adding those this weekend unless someone beats me to it.

Part of my issue is that I think this whole function should be deprecated. I’ll see how painful that will be.

I think I will just always set team.AccessMode=NoAccess whenever team.AccessMode < AdminAccess.

Essentially, every auth middleware should be checking unit permissions for anything less than admin.